### ISSUE TITLE
smartconfig fail on Android 5.0 system

### ISSUE THREAD

**Issue by awong1900** [EspressifApp/EsptouchForAndroid#2]
Hi, guy

I do it on Android 5.0, such as nexus 7(2013) . It often fail.  
But under 5.0, such as nexus 7 (4.3) or note 2(4.0) ,  it is good.

**Comment by EspressifApp** [comment]
Hi Guy,

What version are you using?

Could you send us the log when fail?  

BR,
Baihua

**Comment by awong1900** [comment]
The Android Adb Log?

**Comment by EspressifApp** [comment]
Yes.
And the Esptouch version you are using, the Smart Config version you are using.


--  
baihua
Sent with Sparrow (http://www.sparrowmailapp.com/?sig)

**Comment by awong1900** [comment]
Esptouch version: v3.3
smart config version: v2.2

This is log:
```java
09-01 15:12:41.688  17484-17511/cc.seeed.iot I/Adreno-EGL﹕ <qeglDrvAPI_eglInitialize:379>: QUALCOMM Build: 01/15/15, ab0075f, Id3510ff6dc
09-01 15:12:41.688  17484-17511/cc.seeed.iot I/OpenGLRenderer﹕ Initialized EGL, version 1.4
09-01 15:20:23.791  17484-17511/cc.seeed.iot D/OpenGLRenderer﹕ endAllStagingAnimators on 0xb49b9c80 (NavigationMenuView) with handle 0x9cc60d40
09-01 15:20:44.924  17484-17484/cc.seeed.iot D/IOT﹕ mBtnConfirm is clicked, mEdtApSsid = stu.se.private,  mEdtApPassword = depot0510se
09-01 15:20:44.968  17484-26765/cc.seeed.iot D/UDPSocketServer﹕ mServerSocket is created, socket read timeout: 58000, port: 18266
09-01 15:20:44.968  17484-26765/cc.seeed.iot D/EsptouchTask﹕ execute()
09-01 15:20:44.972  17484-26765/cc.seeed.iot I/EsptouchTask﹕ localInetAddress: /192.168.18.151
09-01 15:20:44.978  17484-26765/cc.seeed.iot D/EsptouchTask﹕ send gc code
09-01 15:20:44.986  17484-17176/cc.seeed.iot D/EsptouchTask﹕ __listenAsyn() start
09-01 15:20:44.998  17484-17176/cc.seeed.iot I/EsptouchTask﹕ expectOneByte: 34
09-01 15:20:44.998  17484-17176/cc.seeed.iot D/UDPSocketServer﹕ receiveSpecLenBytes() entrance: len = 11
09-01 15:20:51.099  17484-26765/cc.seeed.iot D/EsptouchTask﹕ send gc code
09-01 15:20:57.177  17484-26765/cc.seeed.iot D/EsptouchTask﹕ send gc code
09-01 15:21:03.287  17484-26765/cc.seeed.iot D/EsptouchTask﹕ send gc code
09-01 15:21:09.367  17484-26765/cc.seeed.iot D/EsptouchTask﹕ send gc code
09-01 15:21:15.417  17484-26765/cc.seeed.iot D/EsptouchTask﹕ send gc code
09-01 15:21:21.487  17484-26765/cc.seeed.iot D/EsptouchTask﹕ send gc code
09-01 15:21:27.548  17484-26765/cc.seeed.iot D/EsptouchTask﹕ send gc code
09-01 15:21:43.016  17484-17176/cc.seeed.iot W/System.err﹕ java.net.SocketTimeoutException
09-01 15:21:43.026  17484-17176/cc.seeed.iot W/System.err﹕ at libcore.io.IoBridge.maybeThrowAfterRecvfrom(IoBridge.java:598)
09-01 15:21:43.026  17484-17176/cc.seeed.iot W/System.err﹕ at libcore.io.IoBridge.recvfrom(IoBridge.java:556)
09-01 15:21:43.027  17484-17176/cc.seeed.iot W/System.err﹕ at java.net.PlainDatagramSocketImpl.doRecv(PlainDatagramSocketImpl.java:163)
09-01 15:21:43.027  17484-17176/cc.seeed.iot W/System.err﹕ at java.net.PlainDatagramSocketImpl.receive(PlainDatagramSocketImpl.java:171)
09-01 15:21:43.027  17484-17176/cc.seeed.iot W/System.err﹕ at java.net.DatagramSocket.receive(DatagramSocket.java:274)
09-01 15:21:43.027  17484-17176/cc.seeed.iot W/System.err﹕ at cc.seeed.iot.esptouch.udp.UDPSocketServer.receiveSpecLenBytes(UDPSocketServer.java:113)
09-01 15:21:43.027  17484-17176/cc.seeed.iot W/System.err﹕ at cc.seeed.iot.esptouch.task.__EsptouchTask$1.run(__EsptouchTask.java:163)
09-01 15:21:43.027  17484-17176/cc.seeed.iot W/System.err﹕ Caused by: android.system.ErrnoException: recvfrom failed: EAGAIN (Try again)
09-01 15:21:43.027  17484-17176/cc.seeed.iot W/System.err﹕ at libcore.io.Posix.recvfromBytes(Native Method)
09-01 15:21:43.027  17484-17176/cc.seeed.iot W/System.err﹕ at libcore.io.Posix.recvfrom(Posix.java:185)
09-01 15:21:43.028  17484-17176/cc.seeed.iot W/System.err﹕ at libcore.io.BlockGuardOs.recvfrom(BlockGuardOs.java:250)
09-01 15:21:43.028  17484-17176/cc.seeed.iot W/System.err﹕ at libcore.io.IoBridge.recvfrom(IoBridge.java:553)
09-01 15:21:43.028  17484-17176/cc.seeed.iot W/System.err﹕ ... 5 more
09-01 15:21:43.028  17484-17176/cc.seeed.iot I/EsptouchTask﹕ receive rubbish message, just ignore
09-01 15:21:43.028  17484-17176/cc.seeed.iot D/UDPSocketServer﹕ receiveSpecLenBytes() entrance: len = 11
09-01 15:21:43.047  17484-26765/cc.seeed.iot I/UDPSocketClient﹕ USPSocketClient is interrupt
09-01 15:21:43.048  17484-26765/cc.seeed.iot I/UDPSocketServer﹕ USPSocketServer is interrupt
09-01 15:21:43.048  17484-26765/cc.seeed.iot E/UDPSocketServer﹕ mServerSocket is closed
09-01 15:21:43.055  17484-17176/cc.seeed.iot W/System.err﹕ java.net.SocketException: Socket closed
09-01 15:21:43.055  17484-17176/cc.seeed.iot W/System.err﹕ at libcore.io.Posix.recvfromBytes(Native Method)
09-01 15:21:43.055  17484-17176/cc.seeed.iot W/System.err﹕ at libcore.io.Posix.recvfrom(Posix.java:185)
09-01 15:21:43.055  17484-17176/cc.seeed.iot W/System.err﹕ at libcore.io.BlockGuardOs.recvfrom(BlockGuardOs.java:250)
09-01 15:21:43.055  17484-17176/cc.seeed.iot W/System.err﹕ at libcore.io.IoBridge.recvfrom(IoBridge.java:553)
09-01 15:21:43.055  17484-17176/cc.seeed.iot W/System.err﹕ at java.net.PlainDatagramSocketImpl.doRecv(PlainDatagramSocketImpl.java:163)
09-01 15:21:43.055  17484-17176/cc.seeed.iot W/System.err﹕ at java.net.PlainDatagramSocketImpl.receive(PlainDatagramSocketImpl.java:171)
09-01 15:21:43.055  17484-17176/cc.seeed.iot W/System.err﹕ at java.net.DatagramSocket.receive(DatagramSocket.java:274)
09-01 15:21:43.056  17484-17176/cc.seeed.iot W/System.err﹕ at cc.seeed.iot.esptouch.udp.UDPSocketServer.receiveSpecLenBytes(UDPSocketServer.java:113)
09-01 15:21:43.056  17484-17176/cc.seeed.iot W/System.err﹕ at cc.seeed.iot.esptouch.task.__EsptouchTask$1.run(__EsptouchTask.java:163)
09-01 15:21:43.056  17484-17176/cc.seeed.iot I/EsptouchTask﹕ receive rubbish message, just ignore
09-01 15:21:43.061  17484-17176/cc.seeed.iot D/EsptouchTask﹕ __listenAsyn() finish
```

**Comment by EspressifApp** [comment]
Hi Guy,

Are you using the original Smart Config Code and Esptouch Code?  

BR,
Baihua
Sent with Sparrow (http://www.sparrowmailapp.com/?sig)

**Comment by awong1900** [comment]
No , This log be printed by my app. I port all original code to my project.  
And I tested original Smart Config app. Same issue.

I want to know can you test this problem on your side? If not,  Maybe I should check the router or other problems.

**Comment by EspressifApp** [comment]
Hi Guy,

In my opinion, some AP may think that ESPTouch is attacking the AP.

So it will kick the phone using ESPTouch sometimes. To fix the situation, please use ESPTouch v0.3.4 and Smart Config v2.4.  

BR,
Baihua
Sent with Sparrow (http://www.sparrowmailapp.com/?sig)
